test/test_add_separate.mjs

```diff
@@ -0,0 +1,121 @@
+import { getNodeLogs } from '../src/core/helpers.mjs'
+import { NodeCommand } from '../src/commands/node.mjs'
+
+export function testNodeAddSeparated (localBuildPath
+) {
+  describe('Node add via separated commands should success', () => {
+    const suffix = localBuildPath.substring(0, 5)
+    const defaultTimeout = 120000
+    const namespace = 'node-add-separated' + suffix
+    const argv = getDefaultArgv()
+    argv[flags.nodeIDs.name] = 'node1,node2,node3'
+    argv[flags.generateGossipKeys.name] = true
+    argv[flags.generateTlsKeys.name] = true
+    // set the env variable SOLO_FST_CHARTS_DIR if developer wants to use local FST charts
+    argv[flags.chartDirectory.name] = process.env.SOLO_FST_CHARTS_DIR ? process.env.SOLO_FST_CHARTS_DIR : undefined
+    argv[flags.releaseTag.name] = HEDERA_PLATFORM_VERSION_TAG
+    argv[flags.namespace.name] = namespace
+    argv[flags.force.name] = true
+    argv[flags.persistentVolumeClaims.name] = true
+    argv[flags.localBuildPath.name] = localBuildPath
+
+    const argvPrepare = Object.assign({}, argv)
+
+    const tempDir = 'contextDir'
+    argvPrepare[flags.outputDir.name] = tempDir
+
+    const argvExecute = getDefaultArgv()
+    argvExecute[flags.inputDir.name] = tempDir
+    argvExecute[flags.localBuildPath.name] = localBuildPath
+
+    const bootstrapResp = bootstrapNetwork(namespace, argv)
+    const nodeCmd = bootstrapResp.cmd.nodeCmd
+    const accountCmd = bootstrapResp.cmd.accountCmd
+    const networkCmd = bootstrapResp.cmd.networkCmd
+    const k8 = bootstrapResp.opts.k8
+    let existingServiceMap
+    let existingNodeIdsPrivateKeysHash
+
+    afterAll(async () => {
+      await getNodeLogs(k8, namespace)
+      await nodeCmd.accountManager.close()
+      await nodeCmd.stop(argv)
+      await networkCmd.destroy(argv)
+      await k8.deleteNamespace(namespace)
+    }, 600000)
+
+    it('cache current version of private keys', async () => {
+      existingServiceMap = await nodeCmd.accountManager.getNodeServiceMap(namespace)
+      existingNodeIdsPrivateKeysHash = await getNodeIdsPrivateKeysHash(existingServiceMap, namespace, k8, getTmpDir())
+    }, defaultTimeout)
+
+    it('should succeed with init command', async () => {
+      const status = await accountCmd.init(argv)
+      expect(status).toBeTruthy()
+    }, 450000)
+
+    it('should add a new node to the network via the segregated commands successfully', async () => {
+      await nodeCmd.addPrepare(argvPrepare)
+      await nodeCmd.addSubmitTransactions(argvExecute)
+      await nodeCmd.addExecute(argvExecute)
+      expect(nodeCmd.getUnusedConfigs(NodeCommand.ADD_CONFIGS_NAME)).toEqual([
+        flags.app.constName,
+        flags.chainId.constName,
+        flags.devMode.constName,
+        flags.generateGossipKeys.constName,
+        flags.generateTlsKeys.constName,
+        flags.gossipEndpoints.constName,
+        flags.grpcEndpoints.constName,
+        flags.adminKey.constName,
```

Comment:
nitpick: sort alphabetically

Reply:
same as above, it fails if they are sorted

Reply:
oh, yes, that is because adminKey is an extra and not a flag.  I didn't realize that.